tests/integration/functional/zen_stores/test_zen_store.py

```diff
@@ -0,0 +1,991 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import uuid
+from contextlib import ExitStack as does_not_raise
+
+import pytest
+
+from tests.integration.functional.zen_stores.utils import (
+    ComponentContext,
+    CrudTestConfig,
+    PipelineRunContext,
+    RoleContext,
+    StackContext,
+    TeamContext,
+    UserContext,
+    list_of_entities,
+    sample_name,
+)
+from zenml.client import Client
+from zenml.enums import StackComponentType, StoreType
+from zenml.exceptions import (
+    EntityExistsError,
+    IllegalOperationError,
+    StackExistsError,
+)
+from zenml.models import (
+    ArtifactFilterModel,
+    ComponentFilterModel,
+    ComponentUpdateModel,
+    PipelineRunFilterModel,
+    RoleFilterModel,
+    RoleRequestModel,
+    RoleUpdateModel,
+    StackFilterModel,
+    StackRequestModel,
+    StackUpdateModel,
+    StepRunFilterModel,
+    TeamRequestModel,
+    TeamRoleAssignmentRequestModel,
+    TeamUpdateModel,
+    UserRequestModel,
+    UserRoleAssignmentRequestModel,
+    UserUpdateModel,
+    WorkspaceFilterModel,
+    WorkspaceUpdateModel,
+)
+from zenml.models.base_models import (
+    WorkspaceScopedRequestModel,
+)
+from zenml.zen_stores.base_zen_store import (
+    DEFAULT_ADMIN_ROLE,
+    DEFAULT_GUEST_ROLE,
+    DEFAULT_STACK_NAME,
+    DEFAULT_USERNAME,
+    DEFAULT_WORKSPACE_NAME,
+)
+
+DEFAULT_NAME = "default"
+
+# .--------------.
+# | GENERIC CRUD |
+# '--------------'
+
+
+@pytest.mark.parametrize(
+    "crud_test_config",
+    list_of_entities,
+    ids=[e.entity_name for e in list_of_entities],
+)
+def test_basic_crud_for_entity(crud_test_config: CrudTestConfig):
+    """Tests the basic crud operations for a given entity."""
+    client = Client()
+
+    # Create the entity
+    create_model = crud_test_config.create_model
+    if isinstance(create_model, WorkspaceScopedRequestModel):
+        create_model.user = client.active_user.id
+        create_model.workspace = client.active_workspace.id
+    # Test the creation
+    created_entity = crud_test_config.create_method(create_model)
+    # Filter by name to verify the entity was actually created
+    entities_list = crud_test_config.list_method(
+        crud_test_config.filter_model(name=create_model.name)
+    )
+    assert entities_list.total > 0
```

Comment:
```suggestion
    assert entities_list.total == 1
```

Because you only create one, right ? And the name is random, so even if you run it in parallel with other tests on the same or on other machines, it will work.